src/Compilers/CSharp/Test/Symbol/Symbols/Source/LocalTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
+
+public class LocalTests : SemanticModelTestBase
+{
+    [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+    [Fact]
+    public void ForEach()
+    {
+        var sourceCode = @"
+class C
+{
+    void M()
+    {
+        var a = new int[] { 1, 2, 3 };
+        foreach (var x in a)
+        {
+            var y = /*<bind>*/x/*</bind>*/;
+        }
+    }
+}
+";
+        var compilation = CreateCompilation(sourceCode);
+        var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(compilation);
+
+        var symbol = (ILocalSymbol)semanticInfo.Symbol;
+        Assert.False(symbol.IsUsing);
+        Assert.True(symbol.IsForEach);
+        Assert.False(symbol.IsForEach()); // VB extension method works only for VB locals
+    }
+
+    [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+    [Fact]
+    public void ForEachVisualBasic()
+    {
+        var sourceCode = @"
+Module M1
+    Class C
+        Sub S() As Object
+            Dim a() As Integer = {1, 2, 3}
+            For Each x As Integer In a
+                Dim y = x   'BIND:""x""
+            Next
+        End Sub
+    End Class
+End Module
+";
+        var compilation = CreateVisualBasicCompilation(sourceCode);
+        var model = compilation.GetSemanticModel(compilation.SyntaxTrees[0]);
+        var expressionSyntax = VisualBasic.UnitTests.CompilationUtils.FindBindingText<VisualBasic.Syntax.IdentifierNameSyntax>(compilation);
```

Comment:
>FindBindingText [](http://example.com/codeflow?start=70&length=15)

We shouldn't need this method (and the associated .vbproj project reference) if this is the only caller. Consider using something like the following instead:
```csharp
var tree = compilation.SyntaxTrees[0];
var expressionSyntax = tree.GetRoot().DescendantNodes().OfType<VisualBasic.Syntax.IdentifierNameSyntax>().Last();
Assert.Equal("x", expressionSyntax.ToString());
```

Reply:
Alright, thanks. 15c7a861a2836d006cfc4f2ea9c64ad04c1b67e1